script/outside.js

```diff
@@ -0,0 +1,706 @@
+		return $SM.get('game.buildings["hut"]', true) * Outside._HUT_ROOM;
+	},
+	
+	increasePopulation: function() {
+		var space = Outside.getMaxPopulation() - $SM.get('game.population');
+		if(space > 0) {
+			var num = Math.floor(Math.random()*(space/2) + space/2);
+			if(num === 0) num = 1;
+			if(num == 1) {
+				Notifications.notify(null, _('a stranger arrives in the night'));
+			} else if(num < 5) {
+				Notifications.notify(null, _('a weathered family takes up in one of the huts.'));
+			} else if(num < 10) {
+				Notifications.notify(null, _('a small group arrives, all dust and bones.'));
+			} else if(num < 30) {
+				Notifications.notify(null, _('a convoy lurches in, equal parts worry and hope.'));
+			} else {
+				Notifications.notify(null, _("the town's booming. word does get around."));
+			}
+			Engine.log('population increased by ' + num);
+			$SM.add('game.population', num);
+		}
+		Outside.schedulePopIncrease();
+	},
+	
+	killVillagers: function(num) {
+		$SM.add('game.population', num * -1);
+		if($SM.get('game.population') < 0) {
+			$SM.set('game.population', 0);
+		}
+		var remaining = Outside.getNumGatherers();
+		if(remaining < 0) {
+			var gap = -remaining;
+			for(var k in $SM.get('game.workers')) {
+				var numWorkers = $SM.get('game.workers["'+k+'"]');
+				if(numWorkers < gap) {
+					gap -= numWorkers;
+					$SM.set('game.workers["'+k+'"]', 0);
+				} else {
+					$SM.add('game.workers["'+k+'"]', gap * -1);
+					break;
+				}
+			}
+		}
+	},
+	
+	destroyHuts: function(num, allowEmpty) {
+		var dead = 0;
+		for(var i = 0; i < num; i++){
+			var population = $SM.get('game.population', true);
+			var rate = population / Outside._HUT_ROOM;
+			var full = Math.floor(rate);
+			// by default this is used to destroy full or half-full huts
+			// pass allowEmpty to include empty huts in the armageddon
+			var huts = (allowEmpty) ? $SM.get('game.buildings["hut"]', true) : Math.ceil(rate);
+			if(!huts) {
+				break;
+			}
+			// random can be 0 but not 1; however, 0 as a target is useless
+			var target = Math.floor(Math.random() * huts) + 1;
+			var inhabitants = 0;
+			if(target <= full){
+				inhabitants = Outside._HUT_ROOM;
+			} else if(target == full + 1){
+				inhabitants = population % Outside._HUT_ROOM;
+			}
+			$SM.set('game.buildings["hut"]', ($SM.get('game.buildings["hut"]') - 1));
+			if(inhabitants){
+				Outside.killVillagers(inhabitants);
+				dead += inhabitants;
+			}
+		}
+		// this method returns the total number of victims, for further actions
+		return dead;
+	},
+	
+	schedulePopIncrease: function() {
+		var nextIncrease = Math.floor(Math.random()*(Outside._POP_DELAY[1] - Outside._POP_DELAY[0])) + Outside._POP_DELAY[0];
+		Engine.log('next population increase scheduled in ' + nextIncrease + ' minutes');
+		Outside._popTimeout = Engine.setTimeout(Outside.increasePopulation, nextIncrease * 60 * 1000);
+	},
+	
+	updateWorkersView: function() {
+		var workers = $('div#workers');
```

Comment:
The function is potentially a bloater since it has nearly 100 lines of code. Parts of the function could be broken up in to other functions to improve readability